demos/browser/script/webpack-wrapper.js

```diff
@@ -18,11 +18,17 @@ const options = {
   },
 };
 
+const serve = argv.indexOf('--serve') !== -1;
+
+console.info('Should start local server:', serve);
+
 // This won't be present with npm 6.
 // Exclude file paths.
-const lastIndex = argv.length - 1;
+const remainingArgs = argv.filter(v => v !== '--serve');
+const lastIndex = args.length - 1;
+let possibleApp;
 if (lastIndex >= 2) {
-  let possibleApp = argv[lastIndex];
+  possibleApp = remainingArgs[lastIndex];
   if (!possibleApp.startsWith('/')) {
     if (possibleApp.startsWith('app=')) {
       possibleApp = possibleApp.substring(4);
```

Comment:
Should we only set `options.env.npm_config_app = possibleApp;` inside the if?

Reply:
I chose to set it at point of use when we are about to reuse this execution for the server.

Reply:
So what is the difference between `options.env.npm_config_app` and `process.env.npm_config_app`? You set `process.env.npm_config_app` down below but inside this if you set `options.env.npm_config_app`.

Reply:
`options` is the dictionary we use when spawning the Webpack process. `options.env` sets an environment variable for the spawned process.

`process.env` is the env for the _current_ process, so we need to change that before we `require` the server process.

Reply:
Sorry, i meant to ask whether this line https://github.com/aws/amazon-chime-sdk-js/blob/master/demos/browser/script/webpack-wrapper.js#L30 should be set only if `possibleApp.startsWith('app=')`?

Reply:
```js
if (lastIndex >= 2) {
  possibleApp = remainingArgs[lastIndex];
  if (!possibleApp.startsWith('/')) {
    if (possibleApp.startsWith('app=')) {
      possibleApp = possibleApp.substring(4);
    }
    options.env.npm_config_app = possibleApp;
  }
}
```

The first and second `if`s won't be hit unless there's a possible app — the arg count check is for that purpose.

The third `if` strips any leading `key=`.

Webpack is always invoked, so as soon as we know what the possible app is, we set it in the options field.

Does that answer your question?

Reply:
I don't set the `process` value until we know whether we're going to start the server, after `webpack` is done. Setting it to `undefined` is harmless.